### DISCUSSION TITLE
how Upsert document

### DISCUSSION THREAD

**Discussion by genachka** [ravendb/ravendb#14766] [Q&A] [answered]
I've searched the latest documentation and perhaps I missed something but didn't find a way to do an Upsert operation using the Java SDK. Of course I can do a lookup to see if the document exists and if not insert it but that's two operations going back and forth so hoping for an single optimized operation to do that with an Upsert approach.

**Comment by ayende** [comment, marked as answer]
Just call `store`, it will insert / update, as needed.

**Reply by genachka** [reply]
Thanks. I'm already using `store` for insert but hadn't realized that it supports an update operation as needed. This is great!
